subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/IncrementalInputChanges.java

```diff
@@ -16,21 +16,21 @@
 
 package org.gradle.internal.execution.history.changes;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
 import org.gradle.api.tasks.incremental.InputFileDetails;
 import org.gradle.internal.Cast;
 import org.gradle.internal.change.Change;
 import org.gradle.internal.change.CollectingChangeVisitor;
 
-import java.util.Map;
-
 public class IncrementalInputChanges implements InputChangesInternal {
 
     private final InputFileChanges changes;
-    private final Map<Object, String> propertyNameByValue;
+    private final ImmutableListMultimap<Object, String> propertyNamesByValue;
```

Comment:
Do we have to be this specific? Isn't just `ImmutableMultimap` enough?

Reply:
I changed it to `ImmutableMultimap`. I used `ImmutableListMultimap` to be able to call `get(0)` and avoid creating an `Iterator` for the collection. Shouldn't matter, though.